test/001_highsnr_stdio_audio/test_tx.py

```diff
@@ -0,0 +1,141 @@
+ #!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+
+import ctypes
+from ctypes import *
+import pathlib
+import pyaudio
+import time
+import threading
+import audioop
+import argparse
+import sys
+sys.path.insert(0,'..')
+import codec2
+        
+
+# GET PARAMETER INPUTS  
+parser = argparse.ArgumentParser(description='Simons TEST TNC')
+parser.add_argument('--bursts', dest="N_BURSTS", default=0, type=int)
+parser.add_argument('--framesperburst', dest="N_FRAMES_PER_BURST", default=0, type=int)
+parser.add_argument('--delay', dest="DELAY_BETWEEN_BURSTS", default=0, type=int)
+parser.add_argument('--mode', dest="FREEDV_MODE", type=str, choices=['datac0', 'datac1', 'datac3'])
+parser.add_argument('--audiooutput', dest="AUDIO_OUTPUT", default=0, type=int) 
+
+args = parser.parse_args()
+
+N_BURSTS = args.N_BURSTS
+N_FRAMES_PER_BURST = args.N_FRAMES_PER_BURST
+DELAY_BETWEEN_BURSTS = args.DELAY_BETWEEN_BURSTS/1000
+AUDIO_OUTPUT_DEVICE = args.AUDIO_OUTPUT
+
+if args.FREEDV_MODE == "datac0":
+    MODE = codec2.api.FREEDV_MODE_DATAC0
+if args.FREEDV_MODE == "datac1":
+    MODE = codec2.api.FREEDV_MODE_DATAC1
+if args.FREEDV_MODE == "datac1":
+    MODE = codec2.api.FREEDV_MODE_DATAC3
+
```

Comment:
@DJ2LS :point_up: - is it OK if we move to meaningful string based mode names?  There is probably a more pythonic way to convert the strings to constants.  :thinking: - this is also code that would be useful to place in a shared library.

Reply:
Yes I agree!
Maybe we could do this with [enum](https://docs.python.org/3/howto/enum.html). I'll have a look the next days and add it to the todo list.
Are you talking about the codec2 api and something like this?
` mode = freedv_get_mode('datac0')`

Reply:
This should be solvable by moving it to the codec2 module where we are storing the header information 🤔

Reply:
Yes a function like that would be very useful.